src/site/markdown/BndBuild.md

```diff
@@ -51,6 +52,55 @@ If you want to further configure the build can be done in these ways:
 2. You can define properties per project properties in the `bnd.bnd` file `pom.model.property.<some property>: true`, see [the wiki](https://github.com/eclipse-tycho/tycho/wiki/Tycho-Pomless#overwrite-group-and-artifact-ids) for more details.
 3. You can place a `pom.xml` in your `cnf` folder this will then be used as a parent for the aggregator, here you can add additional mojos, profiles and so on. If you want to enable certain things only for some of the projects you can use properties as described in (2) to skip the execution of mojos not relevant for other projects.
 
+#### Create executable jar
+
+Tycho can also create an executable .jar file of your application based on a `.bndrun` file containing `-runbundles`, which you can then deploy and start.
+
+Just add the following line to your `.mvn/maven.config` file or pass it on the maven commandline:
+
+```
+-Dbndrun.exports=mydemo.app
+```
+
+This automatically searches all your bundles for a file named `mydemo.app.bndrun`.
+
+If you have a bundle `myappbundle` which contains `mydemo.app.bndrun`, then the build would create an executable jar at:
+
+`/myappbundle/target/executable/mydemo.app.bndrun.jar`
+
+Your build output looks like this:
+
+```
+[INFO] --- tycho-bnd:4.0.11:run (build) @ myappbundle ---
+[INFO] Exporting mydemo.app.bndrun ...
+[INFO] Exported to .../myappbundle/target/executable/tycho.demo.app.jar
+```
+
+
 ## Mixed Builds
 
 You can even combine a BND Workspace and PDE bundles in a build, see [demo](https://github.com/eclipse-tycho/tycho/tree/master/demo/bnd-pde-workspace).
+
+
+## Troubleshooting
+
+Note: All properties in `.mvn/maven.config` can also be supplied via command line, for example to use a different tycho version use:
+
+`mvn clean install -Dtycho-version=5.0.0-SNAPSHOT`
+
+This uses the snapshot build which is e.g. useful if you build tycho yourself on your local machine and want to test that build.
+
+
+### polyglot.dump.pom
+
+E.g. `-Dpolyglot.dump.pom=pom-gen.xml` specifies that the generated pom.xml files which tycho polyglot creates is called `pom-gen.xml`. 
+This might be needed if you have name clashes with existing files in your build.
+
+
+### tycho.pomless.aggregator.names
+
+`-Dtycho.pomless.aggregator.names=_dummydisabled_` is for handling a rare edge case: 
+In case you have your bnd workspace in a subfolder called one of `bundles`,`plugins`,`tests`,`features`,`sites`,`products`,`releng` then tycho can have problems.
```

Comment:
```suggestion
In case you have your bnd workspace in a subfolder called named `bundles`, `plugins`, `tests`,`features`, `sites`,`products`, or `releng` then tycho can have problems.
```